File: src/dataset/Java/proverb_exercism-uzilan/claude/ClaudeSonnet4_Proverb_v4.java
import java.util.Arrays;

class Proverb {

    private final String[] words;

    Proverb(final String[] words) {
        this.words = Arrays.copyOf(words, words.length);
    }

    String recite() {
        if (words.length == 0) {
            return "";
        }

        final StringBuilder sb = new StringBuilder(estimateCapacity());
        
        for (int i = 0; i < words.length - 1; i++) {
            sb.append("For want of a ").append(words[i])
              .append(" the ").append(words[i + 1])
              .append(" was lost.\n");
        }
        
        sb.append("And all for the want of a ").append(words[0]).append(".");
        return sb.toString();
    }

    private int estimateCapacity() {
        int capacity = 30 + words[0].length();
        for (int i = 0; i < words.length - 1; i++) {
            capacity += 25 + words[i].length() + words[i + 1].length();
        }
        return capacity;
    }
}

File: src/dataset/javascript/proverb_exercism-ThomasZumsteg/claude/ClaudeSonnet4_proverb_v4.js
class Proverb {
    constructor(...args) {
        this.words = [];
        this.qualifier = null;
        
        for (let i = 0; i < args.length; i++) {
            const arg = args[i];
            if (typeof arg === 'string') {
                this.words.push(arg);
            } else if (typeof arg === 'object' && arg && arg.qualifier) {
                this.qualifier = arg.qualifier;
            }
        }
    }

    toString() {
        const wordsLength = this.words.length;
        if (wordsLength === 0) return '';
        
        const result = new Array(wordsLength);
        
        for (let i = 1; i < wordsLength; i++) {
            result[i - 1] = `For want of a ${this.words[i - 1]} the ${this.words[i]} was lost.`;
        }
        
        result[wordsLength - 1] = `And all for the want of a ${this.qualifier ? this.qualifier + ' ' : ''}${this.words[0]}.`;
        
        return result.join('\n');
    }
}

module.exports = Proverb;